src/linklabel.rs

```diff
@@ -34,22 +34,17 @@ pub type LinkLabel<'a> = UniCase<CowStr<'a>>;
 
 /// Assumes the opening bracket has already been scanned.
 /// Returns the number of bytes read (including closing bracket) and label on success.
-pub(crate) fn scan_link_label_rest<F>(
-    text: &str,
-    start: usize,
-    linebreak_handler: F,
-) -> Option<(usize, CowStr<'_>)>
-where
-    // Takes a _global_ index into the document string, returns how many bytes to skip
-    F: Fn(usize) -> Option<usize>,
-{
+pub(crate) fn scan_link_label_rest<'t>(
+    text: &'t str,
+    linebreak_handler: &dyn Fn(&[u8]) -> Option<usize>,
```

Comment:
Any special reason for going `dyn` here? I expect performance to degrade slightly, but of course I don't know that without measuring.

Also, I think it would be useful to have a one-line comment (in the `///` section) explaining the signature, as it's not immediately obvious when reading the code.

Reply:
It's to reduce code bloat (a few KiB by my measurement) for fairly cold routines (linebreaks inside HTML). Figured it was worth the trade-off here.

I'll add a comment explaining the linebreak handler. It is there to facilitate reusing the code for inline HTML. Inline HTML can contain linebreaks and may need access to some context which isn't part of the function arguments.

As described above, this solution isn't particularly elegant and I plan to separate the routines for inline/ block HTML at some point.